### ISSUE TITLE
"Close" Toolbar Icon Attempts To Close The Wrong File Highlighted in The Doc Switcher Panel

### ISSUE THREAD

**Issue by KhunRoger** [notepad-plus-plus/notepad-plus-plus#9194]
### Description of the Issue

The Close toolbar icon - the icon with a red X on it - attempts to close the wrong file in the Doc Switcher panel, a file that is not highlighted.

### Steps to Reproduce the Issue

In the Doc Switcher panel on the left of the Notepad++ window, I have about 20 files listed. The majority of them have blue icons to the left of their name, indicating that they have been saved and have not been edited since saving.
One file at the bottom of the list has been edited. Its icon is red.

When I highlight a file with a blue icon by clicking on the name in the Doc Switcher panel with the left mouse button, the filename and extension are highlighted in dark blue.

When I click on the Close toolbar icon the file is closed and the filename is removed from the Doc Switcher panel.

The dark blue highlight then moves to the filename just below the one deleted.
If that filename below has a red icon - meaning it has not been saved - I can use the up-arrow key to select a filename further up the list in the Doc Switcher panel. This file has a blue icon.

The dark blue highlight then appears on this filename further up the list and a tooltip appears showing the full path of this file.

### Expected Behavior

When I click on the Close toolbar icon the highlighted file should be closed and its name removed from the panel.

### Actual Behavior

When I click on the Close toolbar icon, a pop-up window appears asking if I want to save the file further down the list which has a red icon, i.e. the Close toolbar icon has not registered that I have now selected (using the up-arrow key) a different file that can be closed.

### Debug Information

Notepad++ v7.9   (64-bit)
Build time : Sep 22 2020 - 03:19:04
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 1909
OS Build : 18363.1198
Current ANSI codepage : 1252
Plugins : Explorer.dll mimeTools.dll NppConverter.dll NppExec.dll NppExport.dll

**Comment by KhunRoger** [comment]
Here's a sequence of pictures to illustrate the problem.

1. Highlight the file "SayStartupBeginning.cmd" and click the "Close" icon to remove it.
![1a Screen Shot 2020-12-01 at 22 36 53](https://user-images.githubusercontent.com/6468036/100767160-04488900-342c-11eb-8609-0ffd38e18cb7.jpg)

2. The file is removed from the list and next file down the list is automatically highlighted. Click the "Close" icon to remove it.
![2a Screen Shot 2020-12-01 at 22 37 21](https://user-images.githubusercontent.com/6468036/100767531-78832c80-342c-11eb-83dc-e334b69c5e68.jpg)

3. The file is removed from the list and next file down the list is automatically highlighted. This file has not been saved so use the up-arrow key to highlight the file above.
![3a Screen Shot 2020-12-01 at 22 37 44](https://user-images.githubusercontent.com/6468036/100767965-edeefd00-342c-11eb-9a24-0aa2d851ce7c.jpg)

4. The file above is highlighted. And the tooltip shows the correct filename.
![4a Screen Shot 2020-12-01 at 22 38 35](https://user-images.githubusercontent.com/6468036/100768166-27276d00-342d-11eb-82a2-1fab6bc63b60.jpg)

5. Click the "Close" icon to remove it. A warning window appears referencing the unsaved file which was not highlighted when the "Close" icon was clicked.
![5a Screen Shot 2020-12-01 at 22 39 07](https://user-images.githubusercontent.com/6468036/100768293-4d4d0d00-342d-11eb-9040-33e0540ff08a.jpg)

[Off-topic: Excellent forum software!]

**Comment by KhunRoger** [comment]
Nobody interested in this bug?
Might as well close it then.

**Comment by sasumner** [comment]
>Might as well close it then.

Notepad++ has a limited number of developers that donate their time.
Just because it hasn't been handled yet doesn't mean that it never will be.
Closing, without anyone having time to evaluate it yet, would seem, well, counterproductive.

**Comment by KhunRoger** [comment]
> > Might as well close it then.
> 
> Notepad++ has a limited number of developers that donate their time.
> Just because it hasn't been handled yet doesn't mean that it never will be.
> Closing, without anyone having time to evaluate it yet, would seem, well, counterproductive.

I apologise. I assumed the evaluation process would be a little quicker.

**Comment by ArkadiuszMichalski** [comment]
Anyone can reproduce this? It works fine for me.

**Comment by SkandaSrinivasan** [comment]
So, I was able to reproduce this. From what I can observe from my own tests, which line up with what @KhunRoger did, the close button acts on the tab which is being shown currently(The active tab on screen), and not whatever is selected in the doc selector via the arrow keys.
The illusion that the close button works on whatever is highlighted on the doc selector, is due to the fact that clicking normally with the mouse to select a file in the doc selector, actually highlights and makes that the current tab, thus making that file the target of the close button. Doing it with the keyboard does not bring the highlighted tab into view(active tab) and thus the close button tries to the close the unsaved file(which is the active tab in the screenshots provided by @KhunRoger as well).

A bit wordy, but let me know if I am being unclear and we need screenshots. Also I am not sure if this is a bug or if it's how the close button was intended to work(i.e indepedent of the file highlighted in the doc selector)
